lib/collection/src/shards/channel_service.rs

```diff
@@ -30,4 +34,43 @@ impl ChannelService {
             self.channel_pool.drop_pool(&uri).await;
         }
     }
+
+    /// Get the REST address for the current peer.
+    pub fn current_rest_address(&self, this_peer_id: PeerId) -> CollectionResult<Url> {
+        // Get local peer URI
+        let local_peer_uri = self
+            .id_to_address
+            .read()
+            .get(&this_peer_id)
+            .cloned()
+            .ok_or_else(|| {
+                CollectionError::service_error(format!(
+                    "Cannot determine REST address, this peer not found in cluster by ID {this_peer_id} ",
+                ))
+            })?;
+
+        // Construct REST URL from URI
+        Ok(Url::parse(&format!(
+            "{}://{}:{}",
```

Comment:
Let's try to utilise some ready-made constructors instead, even if it requires an extra conversion step

Reply:
Resolved in 5c5c2e7ea378ee53d8614e51c320b6abbe60d2b6.

Still uses `Url::parse(uri.to_string())`, but I guess this is the best we can do. It does use `set_port()` now and removed some other obsolete bits.